extensions/amp-byside-content/0.1/amp-byside-content.js

```diff
@@ -0,0 +1,343 @@
+    this.unlisteners_ = [];
+
+    /** @private {?string} */
+    this.iframeSrc_ = null;
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {?Promise} */
+    this.iframePromise_ = null;
+
+    /** @private {string}  */
+    this.webcareZone_ = MAIN_WEBCARE_ZONE_;
+
+    /** @private {string}  */
+    this.webcareId_ = '';
+
+    /** @private {string}  */
+    this.channel_ = '';
+
+    /** @private {string}  */
+    this.lang_ = '';
+
+    /** @private {string}  */
+    this.fid_ = '';
+
+    /** @private {string}  */
+    this.label_ = '';
+
+    /** @private {string}  */
+    this.origin_ = '';
+
+    /** @private {string}  */
+    this.baseUrl_ = '';
+
+    /** @const {function()} */
+    this.boundUpdateSize_ =
+		utils.debounce(this.updateSize_.bind(this), 100);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /**
+   * @param {boolean=} onLayout
+   * @override
+   */
+  preconnectCallback(onLayout) {
+    if (this.iframeSrc_) {
+      this.preconnect.url(this.iframeSrc_, onLayout);
+    }
+  }
+
+  /** @override */
+  buildCallback() {
+    this.webcareId_ = user().assert(
+        this.element.getAttribute('data-webcare-id'),
+        'The data-webcare-id attribute is required for <' + TAG_ + '> %s',
+        this.element);
+
+    this.label_ = user().assert(
+        this.element.getAttribute('data-label'),
+        'The data-label attribute is required for <' + TAG_ + '> %s',
+        this.element);
+
+    this.webcareZone_ = (this.element.getAttribute('data-webcare-zone') ||
+		DEFAULT_WEBCARE_ZONE_);
+    this.channel_ = (this.element.getAttribute('data-channel') || '');
+    this.lang_ = (this.element.getAttribute('data-lang') || DEFAULT_LANG_);
+    this.fid_ = (this.element.getAttribute('data-fid') || '');
+
+    this.origin_ = this.composeOrigin_();
+    this.baseUrl_ = this.origin_ + '/BWA' +
+		encodeURIComponent(this.webcareId_) + '/amp/';
+  }
+
+  /** @override */
+  createPlaceholderCallback() {
+    const placeholder = this.win.document.createElement('div');
+    placeholder.setAttribute('placeholder', '');
+    placeholder.appendChild(this.createBySideLoader_());
+
+    this.applyFillContent(placeholder);
+
+    return placeholder;
+  }
+
+  /** @override */
+  layoutCallback() {
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    this.iframe_ = iframe;
+
+    iframe.name = 'amp_byside_content_iframe' + iframeCount_++;
+    iframe.setAttribute('scrolling', 'no');
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowtransparency', 'true');
+    iframe.setAttribute('allowfullscreen', 'true');
+    iframe.setAttribute('title', this.element.getAttribute('title') || '');
+
+    setStyles(iframe, {
+      'opacity': 0,
+    });
+
+    this.element.appendChild(this.getOverflowElement_());
+    this.applyFillContent(iframe);
+
+    return this.composeSrcUrl_().then(src => {
```

Comment:
@aghassemi: This needs to be reverted, or addressed before next cut.

Due to the PII in `#composeSrcUrl_`, this element **cannot** be pre-renderable.

Reply:
as discussed, this is fine since component is not allowed to prerender. I will send PR for other comments soon.

Reply:
Thanks for your review, we are here to help so as soon as a new PR is available we can code out the suggestions.

Reply:
@bysidedevel3rdparty, this PR is merged, so if you sync to master, you can create another PR with just changes requested by @jridgewell. Would appreciate it.

Reply:
A new PR was created in #13363 which incorporated suggestions from @jridgewell , please follow along the code changes there.